cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -208,6 +219,9 @@ public void deleteNamespace(final Id.Namespace namespaceId)
         // Finally delete namespace from MDS
         store.deleteNamespace(namespaceId);
       }
+    } catch (SchedulerException e) {
+      LOG.warn("Error while deleting namespace {}", namespaceId, e);
+      throw new NamespaceCannotBeDeletedException(namespaceId.getId(), e);
```

Comment:
we are using the same warning message and we throw "NamespaceCannotBeDeletedException" for all the cases, should we just have the exception case then, if we don't have to handle them specially ? @bdmogal